### PR TITLE
Implement meta for getMarkets

### LINKED ISSUE #3089: Implement meta object for `getMarkets`
Implement `meta` object. Also re-add `reportingStates` param to `getMarkets`.

### PR DESCRIPTION
**Pull request by adrake33**
Closes #3089.

### CODE REVIEW

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 1]
```diff
@@ -94,6 +110,18 @@ export interface MarketInfoOutcome {
   volume: string;
 }
 
+export enum MarketInfoReportingState {
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts]: Did these move? They are somewhere else, I thought. The UI references ‘Getters.Markets.MarketReportingState.PreReporting’

File: packages/augur-sdk/src/state/getter/Markets.ts [hunk 2]
```diff
@@ -1265,3 +1292,57 @@ async function formatStakeDetails(db: DB, marketId: Address, stakeDetails: any[]
...
+      categories[marketsResult.category1].children[marketsResult.category2] = {
+        count: 1,
+        children: {},
+      };
+    }
+    if (categories[marketsResult.category1].children[marketsResult.category2].children[marketsResult.category3]) {
+      categories[marketsResult.category1].children[marketsResult.category2].children[marketsResult.category3]['count']++;
+    } else {
+      categories[marketsResult.category1].children[marketsResult.category2].children[marketsResult.category3] = {
+        count: 1,
```
Comment (bthaile) [inline packages/augur-sdk/src/state/getter/Markets.ts:1295]: Does count for category3 ever increment?

**Review by bthaile — APPROVED** [review]
A few questions, will have to wire it up to truly test. UI effort to do so is coming up.

File: packages/augur-sdk/src/state/getter/Markets.ts (hunk 2 above)
Comment (adrake33) [reply packages/augur-sdk/src/state/getter/Markets.ts:1295]: Yep, it gets incremented in the if statement above.

File: packages/augur-sdk/src/state/getter/Markets.ts (hunk 1 above)
Comment (adrake33) [reply packages/augur-sdk/src/state/getter/Markets.ts]: Oh good catch. I'll rename it.

**Pull request merged**